openhcl/underhill_core/src/dma_manager.rs

```diff
@@ -0,0 +1,131 @@
+//use std::sync::{Mutex, Weak};
+
+use memory_range::MemoryRange;
+use std::{collections::HashMap, sync::{Arc, Mutex}};
+use page_pool_alloc::PagePool;
+use user_driver::{memory::MemoryBlock, vfio::VfioDmaBuffer};
+use user_driver::lockmem::LockedMemorySpawner;
+
+#[derive(Clone)]
```

Comment:
Like page pool, implementing `Clone` on this type seems odd to me. You want there to be only a _single_ manager, and make it painfully obvious to the user that you do _not_ want to clone this type because it has special internal state that it is responsible for managing. This means that you may need to pass usage of it by ref, or by Arc<Self>